crates/gen-spidermonkey/src/lib.rs

```diff
@@ -0,0 +1,2180 @@
+            witx2::abi::Instruction::BufferLiftPtrLen { push: _, ty: _ } => todo!(),
+            witx2::abi::Instruction::BufferLiftHandle { push: _, ty: _ } => todo!(),
+            witx2::abi::Instruction::RecordLower {
+                record: _,
+                name: _,
+                ty: _,
+            } => todo!(),
+            witx2::abi::Instruction::RecordLift {
+                record: _,
+                name: _,
+                ty: _,
+            } => todo!(),
+            witx2::abi::Instruction::FlagsLower {
+                record: _,
+                name: _,
+                ty: _,
+            } => todo!(),
+            witx2::abi::Instruction::FlagsLower64 {
+                record: _,
+                name: _,
+                ty: _,
+            } => todo!(),
+            witx2::abi::Instruction::FlagsLift {
+                record: _,
+                name: _,
+                ty: _,
+            } => todo!(),
+            witx2::abi::Instruction::FlagsLift64 {
+                record: _,
+                name: _,
+                ty: _,
+            } => todo!(),
+            witx2::abi::Instruction::VariantPayloadName => todo!(),
+            witx2::abi::Instruction::VariantLower {
+                variant: _,
+                name: _,
+                ty: _,
+                results: _,
+            } => todo!(),
+            witx2::abi::Instruction::VariantLift {
+                variant: _,
+                name: _,
+                ty: _,
+            } => todo!(),
+            witx2::abi::Instruction::CallWasm { module, name, sig } => {
+                // Push the Wasm arguments.
+                //
+                // []
+                let locals: Vec<_> = sig.params.iter().map(|_| pop_wasm(operands)).collect();
+                for local in locals.into_iter().rev() {
+                    self.inst(Instruction::LocalGet(local));
+                }
+                // [A...]
+
+                let func_index = self
+                    .gen
+                    .import_fn_name_to_index
+                    .get(*module)
+                    .unwrap()
+                    .get(*name)
+                    .unwrap()
+                    .0;
+                self.inst(Instruction::Call(func_index));
+                // [R...]
+
+                // Allocate locals for the results and pop the return values off
+                // the Wasm stack, saving each of them to the associated local.
+                let locals: Vec<_> = sig
+                    .results
+                    .iter()
+                    .map(|ty| {
+                        self.new_local(match ty {
+                            WasmType::I32 => wasm_encoder::ValType::I32,
+                            WasmType::I64 => wasm_encoder::ValType::I64,
+                            WasmType::F32 => wasm_encoder::ValType::F32,
+                            WasmType::F64 => wasm_encoder::ValType::F64,
+                        })
```

Comment:
This might make a good helper function (the `match` statement) since I think I saw this somewhere else as well?